test/tests/functional/pbs_hook_exechost_periodic.py

```diff
@@ -74,3 +74,130 @@ def test_multiple_exechost_periodic_hooks(self):
+        self.hostB = self.momB.shortname
+        hook_name = "periodic"
+        hook_attrs = {'event': 'exechost_periodic', 'enabled': 'True'}
+        hook_body = common_periodic_hook_script
+        self.server.create_import_hook(hook_name, hook_attrs, hook_body)
+
+        exp_msg = "In exechost_periodic hook"
+        for mom in self.moms.values():
+            mom.log_match(exp_msg)
+
+        other_node = "invalid_node"
+        common_msg = " as it is owned by a different mom"
+        common_msg2 = "resources_available.mem=9gb per mom hook request"
+
+        msg1 = "%s;Not allowed to update vnode '%s'," % (self.momA.hostname,
+                                                         self.hostB)
+        exp_msg1 = msg1 + common_msg
+        exp_msg2 = "%s;autocreated vnode %s" % (self.momA.hostname,
+                                                other_node)
+        msg2 = "%s;Updated vnode %s's resource " % (self.momA.hostname,
+                                                    other_node)
+        exp_msg3 = msg2 + common_msg2
+        msg3 = "%s;Updated vnode %s's resource " % (self.momA.hostname,
+                                                    other_node)
+        exp_msg4 = msg3 + common_msg2
+        msg4 = "%s;Not allowed to update vnode '%s'," % (self.momB.hostname,
+                                                         other_node)
+        exp_msg5 = msg4 + common_msg
+
+        for msg in [exp_msg1, exp_msg2, exp_msg3, exp_msg4, exp_msg5]:
+            self.server.log_match(msg)
+
+        node_attribs = {'resources_available.mem': "90gb"}
+        self.server.expect(NODE, node_attribs, id=self.momB.shortname)
+
+    @requirements(num_moms=2)
+    def test_exechost_periodic_alarm(self):
+        """
+        Test exechost_periodic with alarm timeout in hook script
+        """
+        hook_name = "periodic"
+        hook_attrs = {'event': 'exechost_periodic', 'enabled': 'True',
+                      'alarm': '5'}
+        hook_script = """time.sleep(10)"""
+        hook_body = """import time \n"""
+        hook_body += common_periodic_hook_script + hook_script
+        self.server.create_import_hook(hook_name, hook_attrs, hook_body)
+        log_msg = "alarm call while running exechost_periodic hook"
+        log_msg += " '%s', request rejected" % hook_name
+        exp_msg = ["In exechost_periodic hook",
+                   log_msg,
+                   "Non-zero exit status 253 encountered for periodic hook",
+                   "exechost_periodic request rejected by '%s'" % hook_name]
+        for mom in self.moms.values():
+            for msg in exp_msg:
+                mom.log_match(msg)
+
+    @requirements(num_moms=2)
+    def test_exechost_periodic_error(self):
+        """
+        Test exechost_periodic with an unhandled exception in the hook script
+        """
+        hook_name = "periodic"
+        hook_attrs = {'event': 'exechost_periodic', 'enabled': 'True'}
+        hook_script = "x"
```

Comment:
Instead of using a syntax error to raise an exception, why not raise one yourself?  It is clearer to the reader.